turbopack/crates/turbopack-node/src/pool.rs

```diff
@@ -793,7 +800,14 @@ impl NodeJsPool {
                 let idle_process_permit = idle_process_permit.context("acquiring idle process permit")?;
                 let process = {
                     let mut processes = self.processes.lock();
-                    processes.pop().unwrap()
+                    let process = processes.pop().unwrap();
+                    {
+                        let mut pools = ACTIVE_POOLS.lock();
+                        if let Some(idx) = pools.iter().position(|p| Arc::ptr_eq(p, &self.processes)) {
+                            pools.swap_remove(idx);
+                        }
```

Comment:
Should this really unconditionally remove it? Doesn't it depend on how many idle processes are still left after acquiring one?

Reply:
oh... good point. It should check for `is_empty()`